src/olympia/files/views.py

```diff
@@ -37,3 +48,48 @@ def serve_file_upload(request, uuid):
     return HttpResponseXSendFile(
         request, upload.path, content_type='application/octet-stream'
     )
+
+
+class FileUploadViewSet(CreateModelMixin, ReadOnlyModelViewSet):
+    queryset = FileUpload.objects.all()
+    serializer_class = FileUploadSerializer
+    permission_classes = [
+        APIGatePermission('addon-submission-api'),
+        AllowOwner,
+        IsSubmissionAllowedFor,
+    ]
+    authentication_classes = [JWTKeyAuthentication, WebTokenAuthentication]
+    lookup_field = 'uuid'
+
+    def get_queryset(self):
+        return super().get_queryset().filter(user=self.request.user)
+
+    def create(self, request):
+        # TODO: move most/all of this logic to the serializer
```

Comment:
Why not just address this `# TODO` now? Is it something that you are planning as part of a follow-up?

Reply:
I'm still not sure if I should make that change - doing things in the serializer is generally the rest framework way, but in this case there is very little that can set or we need to validate, and update functionality isn't appropriate for FileUpload instances. I'll remove the TODO comment, because it's not really a definite TODO any longer.